packages/components/fields/time-field/src/time-field.tsx

```diff
@@ -0,0 +1,263 @@
+   * <br />
+   * It can also return null to show no error.
+   */
+  renderError?: (key: string, error?: boolean) => ReactNode;
+
+  /**
+   * Indicates if the value is required. Shows an the "required asterisk" if so.
+   */
+  isRequired?: boolean;
+
+  /**
+   * Indicates whether the field was touched. Errors will only be shown when the field was touched.
+   */
+  touched?: boolean;
+
+  // TimeInput
+  /**
+   * Used as HTML name of the input component.
+   */
+  name?: string;
+
+  /**
+   * Used as HTML autocomplete of the input component.
+   */
+  autoComplete?: string;
+
+  /**
+   * Value of the input
+   */
+  value: string;
+
+  /**
+   * Called with an event holding the new value.
+   * <br/>
+   * Required when input is not read only. Parent should pass it back as `value`-
+   */
+  onChange: (event: TEvent) => void;
+
+  /**
+   * Called when input is blurred
+   */
+  onBlur?: (event: TEvent) => void;
+
+  /**
+   * Called when input is focused
+   * <br/>
+   * Signature: `(event) => void`
+   */
+  onFocus?: (event: TEvent) => void;
+
+  /**
+   * Focus the input on initial render
+   */
+  isAutofocussed?: boolean;
+
+  /**
+   * Indicates that the input cannot be modified (e.g not authorized, or changes currently saving).
+   */
+  isDisabled?: boolean;
+
+  /**
+   * Indicates that the input is read only (no changes allowed).
+   */
+  isReadOnly?: boolean;
+
+  /**
+   * Placeholder text for the input
+   */
+  placeholder?: string;
+
+  // LabelField
+  /**
+   * Title of the label
+   */
+  title: ReactNode;
+
+  /**
+   * Hint for the label. Provides a supplementary but important information regarding the behaviour of the input (e.g warn about uniqueness of a field, when it can only be set once),
+   * whereas description can describe it in more depth. Can also receive a hintIcon.
+   */
+  hint?: ReactNode;
+
+  /**
+   * Provides a description for the title.
+   */
+  description?: ReactNode;
+
+  /**
+   * Function called when info button is pressed.
+   * <br/>
+   * Info button will only be visible when this prop is passed.
+   */
+  onInfoButtonClick?: (
+    event: MouseEvent<HTMLButtonElement> | KeyboardEvent<HTMLButtonElement>
+  ) => void;
+
+  /**
+   * Icon to be displayed beside the hint text. Will only get rendered when hint is passed as well.
+   */
+  hintIcon?: ReactElement;
+
+  /**
+   * Badge to be displayed beside the label.
+   * <br/>
+   * Might be used to display additional information about the content of the field (E.g verified email)
+   */
+  badge?: ReactNode;
+};
+
+type TTimeFieldState = Pick<TTimeFieldProps, 'id'>;
+
+const sequentialId = createSequentialId('time-field-');
+
+const hasErrors = (errors?: TFieldErrors) =>
+  errors && Object.values(errors).some(Boolean);
+
+class TimeField extends Component<TTimeFieldProps, TTimeFieldState> {
+  static displayName = 'TimeField';
+
+  static defaultProps = {
```

Comment:
It'll be nice to type the defaultProps.

Reply:
Typed with this commit: 9990fac4c1e0c0f9c7f1d6bedde91885131541e3